core/services/relay/evm/target_strategy.go

```diff
@@ -0,0 +1,230 @@
+package evm
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	"github.com/smartcontractkit/chainlink-framework/capabilities/writetarget"
+)
+
+var (
+	_           writetarget.TargetStrategy = &evmTargetStrategy{}
+	ErrTxFailed                            = errors.New("submitted transaction failed")
+)
+
+type evmTargetStrategy struct {
+	cr commontypes.ContractReader
+	cw commontypes.ContractWriter
+
+	lggr      logger.Logger
+	forwarder string
+
+	receiverGasMinimum uint64
+	binding            commontypes.BoundContract
+	bound              *atomic.Bool
+}
+
+type TransmissionInfo struct {
+	GasLimit        *big.Int
+	InvalidReceiver bool
+	State           uint8
+	Success         bool
+	TransmissionID  [32]byte
+	Transmitter     common.Address
+}
+
+// The gas cost of the forwarder contract logic, including state updates and event emission.
+// This is a rough estimate and should be updated if the forwarder contract logic changes.
+// TODO: Make this part of the on-chain capability configuration
+const ForwarderContractLogicGasCost = 100_000
+
+const (
+	TransmissionStateNotAttempted uint8 = iota
+	TransmissionStateSucceeded
+	TransmissionStateInvalidReceiver
+	TransmissionStateFailed
+)
+
+type Config struct {
+	// Address of the contract that will get the forwarded report
+	Address string
```

Comment:
Is there a reason to leave this as config and not move to input? At least two user requests have a use case to dynamically set the address.

Reply:
I am not aware of a reason why this couldn't be an input, but I think this discussion is out of scope for this PR and should be a followup